types/types.ts

```diff
@@ -1,10 +1,25 @@
 import { UserJSON } from '@clerk/backend-core'
-import { Patient, Staff } from '@prisma/client'
+import {
+  Patient,
+  Staff,
+  Prescription,
+  Location,
+  LockerBox
+} from '@prisma/client'
 
 export enum Role {
   Staff = 'staff',
   Patient = 'patient'
 }
+export type PrescriptionAndLocation = Prescription & { Location: Location } & {
+  LockerBox: LockerBox
+}
+
+export type PharmacyLocation = Location & {
+  Prescriptions: Prescription[]
+} & {
+  LockerBoxes: LockerBox[]
+}
```

Comment:
this works! you can also use the typescript `extends` definer https://www.typescriptlang.org/docs/handbook/interfaces.html#extending-interfaces. 

can sometimes make it easier to see the true nested structure of types